app/assets/javascripts/components/pageElements/page_container.js.jsx

```diff
@@ -0,0 +1,64 @@
+var PageContainer = React.createClass({
+  getInitialState: function(){
+    return {categories: [],
+            selectedCategory: {name: 'All', id: 9},
+            loggedIn: false,
+            user: '',
+    }
+  },
+  componentWillMount: function(){
+    this.loadCategoriesFromServer();
+    this.setLoggedInStatus();
+  },
+  setLoggedInStatus: function() {
+    var self = this
+    $.get('/sessions/user').done(function(response){
+      self.setState({loggedIn: true, user: response.username})
+    }).fail()
+  },
+  loadCategoriesFromServer: function(){
+    $.ajax({
+      url: '/categories',
+      dataType: 'json',
+      method: 'GET',
+      success: function(categories){
+        categories.unshift({name: 'All', id: 9})
+        this.setState({categories: categories, selectedCategory: categories[0]});
+      }.bind(this),
+      error: function(xhr,status,err){
+        console.error(this.props.url,status,err.toString())
+      }.bind(this)
+    });
+  },
+  shouldComponentUpdate: function(){
```

Comment:
This defaults to true, so it's not necessary. When might this function be useful?